src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,190 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import core.basesyntax.service.exception.RegisterException;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDaoImpl storageDao;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_userAlreadyExists_notOk() {
+        User user = new User();
+        user.setLogin("ExistsLogin");
+        user.setPassword("ValidPassword");
+        user.setAge(25);
+        storageDao.add(user);
+        assertThrows(RegisterException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void loginIsValid_nullLogin_notOk() {
+        User user = new User();
+        user.setLogin(null);
+        user.setPassword("ValidPassword");
+        user.setAge(25);
+        assertThrows(RegisterException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void loginIsValid_emptyStringLogin_notOk() {
+        User user = new User();
+        user.setLogin("");
+        user.setPassword("ValidPassword");
+        user.setAge(25);
+        assertThrows(RegisterException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void loginIsValid_lessThenSixCharactersLogin_notOk() {

```

Comment:
rename, for example: 
register_loginLessThenMinLength_notOk

If the min length limit will be changed -> you will not need to change the name of test method. 
Check all places